lib/AST/ASTImporter.cpp

```diff
@@ -3001,32 +3001,46 @@ ExpectedDecl ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
       }
 
       if (FunctionDecl *FoundFunction = dyn_cast<FunctionDecl>(Found)) {
-        if (FoundFunction->hasExternalFormalLinkage() &&
-            D->hasExternalFormalLinkage()) {
-          if (IsStructuralMatch(D, FoundFunction)) {
-            const FunctionDecl *Definition = nullptr;
-            if (D->doesThisDeclarationHaveABody() &&
-                FoundFunction->hasBody(Definition))
-              return Importer.MapImported(
-                  D, const_cast<FunctionDecl *>(Definition));
-            FoundByLookup = FoundFunction;
-            break;
-          }
 
-          // FIXME: Check for overloading more carefully, e.g., by boosting
-          // Sema::IsOverload out to the AST library.
+        bool VisibilityMatch = true;
```

Comment:
Could we simplify line 3005-3018 this way?
```
bool VisibilityMatch = D->hasExternalFormalLinkage() == FoundFunction->hasExternalFormalLinkage() && 
  D->isInAnonymousNamespace() == FoundFunction->isInAnonymousNamespace() &&
  Importer.GetFromTU(FoundFunction) == D->getTranslationUnitDecl();
```

Reply:
Also perhaps the whole could go in a separate function to make smaller the already long VisitFunctionDecl.

Reply:
If we experiment with formal logic expressions probably we can find a more simple way to use instead of `if` but the current code may be more easy to understand. Testing for "everything is equal" (like above) is not correct.